wandb/sdk/launch/agent/agent.py

```diff
@@ -357,43 +376,35 @@ async def finish_thread_id(
                 fnames,
             )
         elif job_and_run_status.run is not None:
-            run_info = None
+            called_init = False
             # We do some weird stuff here getting run info to check for a
             # created in run in W&B.
             #
             # We retry for 60 seconds with an exponential backoff in case
             # upsert run is taking a while.
-            #
-            # Sweep runs exist but have no info before they are started
-            # so run_info returned will be None, while normal runs just throw a
-            # comm error.
             logs = None
             start_time = time.time()
             interval = 1
             while True:
-                try:
-                    run_info = self._api.get_run_info(
-                        self._entity,
-                        job_and_run_status.project,
-                        job_and_run_status.run_id,
-                    )
-                except CommError:
-                    pass
-                if (
-                    run_info is not None
-                    or time.time() - start_time > RUN_INFO_GRACE_PERIOD
-                ):
+                assert job_and_run_status.project is not None
```

Comment:
If these assertions ever fail the user will just get a vague, unactionable error.  At the very least, can we put a message for the `AssertionError`

Reply:
I can add a comment and more info here. But this should never happen. I'll try to modify the class to make these not optional.